crates/ruff_python_parser/src/lib.rs

```diff
@@ -518,6 +510,18 @@ impl Deref for Tokens {
     }
 }
 
+impl From<&Tokens> for CommentRanges {
+    fn from(tokens: &Tokens) -> Self {
+        let mut ranges = vec![];
+        for token in tokens {
+            if token.kind() == TokenKind::Comment {
+                ranges.push(token.range());
+            }
+        }
+        CommentRanges::new(ranges)
+    }
+}
+
```

Comment:
I think this is a better approach to create `CommentRanges` instead of `CommentRangesBuilder`. This is the primary way of building the `CommentRanges` outside the linter. In the linter, the `Indexer` creates it because it's already looping over the tokens.